### PR TITLE
Sprint 15 315 wysiwyg component

### LINKED ISSUE #315: Weaver Components should offer a WYSIWYG component.
This should wrap the tiny mce library: 

https://www.tiny.cloud/docs-4x/integrations/angular2/

This should allow for the customization of the editor through `@Input` properties.

Here is an example of the tiny mce editor in angular: 

https://tinymce.github.io/tinymce-angular/?path=/story/editor--iframe-editor

We could use a config json object for wholesale customization.

### PR DESCRIPTION
**Pull request by rmathew1011**
Resolves #315

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,11 +50,11 @@ jobs:
           LHCI_GITHUB_APP_TOKEN: ${{ secrets.LHCI_GITHUB_APP_TOKEN }}
 
       - name: "Deploy GH-Pages"
-        if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+        # if: github.ref == 'refs/heads/master' && github.event_name == 'push'
```
Comment (jeremythuff) [inline .github/workflows/build.yml:53]: Please revert this change

File: angular.json [hunk 2]
```diff
@@ -28,12 +28,15 @@
               "src/favicon.ico",
               "src/assets",
               "src/config.json",
-              "src/weaver-components.js"
+              "src/weaver-components.js",
+              { "glob": "**/*", "input": "node_modules/tinymce", "output": "/tinymce/" }
```
Comment (jeremythuff) [inline angular.json:32]: Ensure this is needed.

File: angular.json [hunk 3]
```diff
@@ -28,12 +28,15 @@
...
             ],
             "styles": [
               "src/styles.scss"
             ],
-            "scripts": [],
+            "scripts": [
+              "node_modules/tinymce/tinymce.min.js"
```
Comment (jeremythuff) [inline angular.json:38]: Ensure this is needed.

File: angular.json [hunk 4]
```diff
@@ -60,7 +63,7 @@
                 {
                   "type": "initial",
                   "maximumWarning": "1.5mb",
-                  "maximumError": "2mb"
+                  "maximumError": "2.1mb"
```
Comment (jeremythuff) [inline angular.json:66]: If we are able to remove the changes above, is the bundle size still too large?

File: angular.json (hunk 2 above)
Comment (jeremythuff) [reply angular.json:32]: https://www.tiny.cloud/docs-4x/integrations/angular2/

File: package.json [hunk 5]
```diff
@@ -76,6 +77,7 @@
     "json5": "^2.1.3",
     "ng-inline-svg": "^11.0.1",
     "rxjs": "~6.6.3",
+    "tinymce": "^5.7.0",
```
Comment (jeremythuff) [inline package.json:80]: Ensure that this is needed.

File: projects/wvr-elements/src/lib/shared/wvr-shared.module.ts [hunk 6]
```diff
@@ -68,9 +72,14 @@ export const WVR_PIPES = [
...
   ],
-  providers: [],
+  providers: [
+    { provide: TINYMCE_SCRIPT_SRC, useValue: 'tinymce/tinymce.min.js' }
+  ],
   entryComponents: [],
-  schemas: [CUSTOM_ELEMENTS_SCHEMA]
+  schemas: [
+    CUSTOM_ELEMENTS_SCHEMA,
+    NO_ERRORS_SCHEMA
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/shared/wvr-shared.module.ts:81]: Ensure this is needed

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+@Component({
+  selector: 'wvr-editor-component',
+  templateUrl: './wvr-editor.component.html',
+  styleUrls: ['./wvr-editor.component.scss']
+})
+export class WvrEditorComponent extends WvrBaseComponent implements AfterViewInit {
+
+  htmlId = `wvr-editor-${this.id}`;
+
+  @Input() apiKey = 'no-api-key';
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:14]: Maybe we could use the cloud version...

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+  @Input() cloudChannel = '5-stable';
+
+  @Input() skinUrl =  `${this.appConfig.assetsUrl}/skins/ui/oxide-dark`;
+
+  @Input() initialValue = '';
+
+  @Input() disabled: 'true' | 'false' = 'false';
+
+  @Input() menuBar = 'file edit view insert format table tools help';
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:24]: Lets use json configuration for this.

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                            'help wordcount print preview spellchecker'
+                          ];
+
+  @Input() outputFormat: "html"|"text" = "text";
+
+  @Input() referrerPolicy = 'strict-origin-when-cross-origin';
+
+  @Input() toolBar = ['undo redo | formatselect | bold italic underline code | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | help | print preview | media spellchecker' ];
+
+  @Input() menu: Object = `{
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:38]: We can, possibly, use a setter and getter for this and JSON5 to parse a string representation of this json. We can also use an interface to express the form of the menu configuration.

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                            edit: { title: 'Edit', items: 'undo redo | cut copy paste | selectall | searchreplace' },
+                            view: { title: 'View', items: 'code | visualaid visualchars visualblocks | spellchecker | preview fullscreen' },
+                            insert: { title: 'Insert', items: 'image link media template codesample inserttable | charmap emoticons hr | pagebreak nonbreaking anchor toc | insertdatetime' },
+                            format: { title: 'Format', items: 'bold italic underline strikethrough superscript subscript codeformat | formats blockformats fontformats fontsizes align lineheight | forecolor backcolor | removeformat' },
+                            tools: { title: 'Tools', items: 'spellchecker spellcheckerlanguage | code wordcount' },
+                            table: { title: 'Table', items: 'inserttable | cell row column | tableprops deletetable' },
+                            help: { title: 'Help', items: 'help'
+                          }`;
+
+  @Input() mobileMenu: Object = `{
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:49]: See above comment

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+      menubar: this.menuBar,
+      toolbar: this.toolBar
+      /* enable title field in the Image dialog*/
+      // image_title: true,
+      /* enable automatic uploads of images represented by blob or data URIs*/
+      // automatic_uploads: true,
+
+    });
+
+    tinymce.init({
```
Comment (jeremythuff) [inline projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:77]: This is most likely the method we will want to use

File: src/assets/skins/content/dark/content.css [hunk 12]
```diff
@@ -0,0 +1,72 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/content/dark/content.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/content/default/content.css [hunk 13]
```diff
@@ -0,0 +1,67 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/content/default/content.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/content/document/content.css [hunk 14]
```diff
@@ -0,0 +1,72 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/content/document/content.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/content/writer/content.css [hunk 15]
```diff
@@ -0,0 +1,68 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/content/writer/content.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide-dark/content.mobile.css [hunk 16]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide-dark/content.mobile.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide-dark/content.mobile.min.css [hunk 17]
```diff
@@ -0,0 +1,7 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide-dark/content.mobile.min.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide-dark/skin.shadowdom.css [hunk 18]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide-dark/skin.shadowdom.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide-dark/skin.shadowdom.min.css [hunk 19]
```diff
@@ -0,0 +1,7 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide-dark/skin.shadowdom.min.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide/content.mobile.css [hunk 20]
```diff
@@ -0,0 +1,29 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide/content.mobile.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide/skin.shadowdom.css [hunk 21]
```diff
@@ -0,0 +1,37 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide/skin.shadowdom.css:1]: These styles should be expressed at the component level if possible.

File: src/assets/skins/ui/oxide/skin.shadowdom.min.css [hunk 22]
```diff
@@ -0,0 +1,7 @@
+/**
```
Comment (jeremythuff) [inline src/assets/skins/ui/oxide/skin.shadowdom.min.css:1]: These styles should be expressed at the component level if possible.

File: src/index.html [hunk 23]
```diff
@@ -885,6 +889,7 @@ <h3><wvre-text value="Second Tab Header"></wvre-text></h3>
     }
   }"></wvre-theme>
 
+  <script defer src="tinymce/tinymce.min.js"></script>
```
Comment (jeremythuff) [inline src/index.html:892]: We need to avoid the global usage of tinymce.

**Review by jeremythuff — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (rmathew1011) [reply .github/workflows/build.yml:53]: Change reverted

File: projects/wvr-elements/src/lib/shared/wvr-shared.module.ts (hunk 6 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/shared/wvr-shared.module.ts:81]: Removed NO_ERRORS_SCHEMA

File: projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts (hunk 8 above)
Comment (rmathew1011) [reply projects/wvr-elements/src/lib/wvr-editor/wvr-editor.component.ts:24]: Utilizing json format for menu instead of Object

File: src/index.html (hunk 23 above)
Comment (rmathew1011) [reply src/index.html]: Removed this import

**Review by jeremythuff — APPROVED** [review]
